### PR TITLE
[CL-135] Migrate component library to standalone components

### PR DESCRIPTION
**Pull request by Hinton**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/CL-135

## 📔 Objective

Angular is making Standalone Components the default in v19. This PR explores the automated migration tool in Angular to evaluate how it behaves, the output which we can use to determine how quickly we desire to migrate if at all.

To migrate run `ng generate @angular/core:standalone`.

Afterwards the converted imports needs to be cleaned up. Angular sometimes finds weird modules to import, for the CL this commonly resulted in:
- Imports to JslibModule which we do not not support in CL.
- Direct imports to component/directives where modules are more suited.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.component.spec.ts [hunk 1]
```diff
@@ -45,13 +45,8 @@ describe("NavigationProductSwitcherComponent", () => {
     mockProducts$.next({ bento: [], other: [] });
 
     await TestBed.configureTestingModule({
-      imports: [RouterModule],
-      declarations: [
-        NavigationProductSwitcherComponent,
-        NavItemComponent,
-        BitIconButtonComponent,
```
Comment (Hinton) [inline apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.component.spec.ts:52]: Bad imports, they should have used the module in the first place.

File: bitwarden_license/bit-web/src/app/tools/access-intelligence/password-health.component.spec.ts [hunk 2]
```diff
@@ -27,7 +26,7 @@ describe("PasswordHealthComponent", () => {
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [PasswordHealthComponent, PipesModule, TableModule, LooseComponentsModule],
-      declarations: [TableBodyDirective],
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/tools/access-intelligence/password-health.component.spec.ts:30]: TableModule already provides this directive and is the correct way to import from the CL. Hence why the import statement contains `src`.

File: libs/components/src/navigation/nav-item.component.ts [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
+import { IconButtonModule } from "../icon-button";
+
 import { NavBaseComponent } from "./nav-base.component";
-import { NavGroupComponent } from "./nav-group.component";
 import { SideNavService } from "./side-nav.service";
 
+// Resolves a circular dependency between `NavItemComponent` and `NavItemGroup` when using standalone components.
+export abstract class NavGroupAbstraction {
+  abstract setOpen(open: boolean): void;
+}
```
Comment (willmartian) [inline libs/components/src/navigation/nav-item.component.ts:14]: 👍

**Review by willmartian — APPROVED** [review]
Excellent, thanks for doing this. The Angular code migrations are impressive--and Storybook showing no diffs grants me confidence that it migrated correctly.

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**